flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/parser/TransformParser.java

```diff
@@ -588,7 +585,8 @@ public static String normalizeFilter(String projection, String filter) {
 
     public static boolean hasAsterisk(@Nullable String projection) {
         if (isNullOrWhitespaceOnly(projection)) {
-            return false;
+            // Providing an empty projection expression is equivalent to writing `*` explicitly.
+            return true;
```

Comment:
Please rebase with `master` as this has been resolved

Reply:
I'll rebase it tonight.